scaler/handlers.go

```diff
@@ -38,18 +38,39 @@ func (e *impl) IsActive(
 	ctx context.Context,
 	scaledObject *externalscaler.ScaledObjectRef,
 ) (*externalscaler.IsActiveResponse, error) {
+	lggr := e.lggr.WithName("IsActive")
+	host, ok := scaledObject.ScalerMetadata["host"]
+	if !ok {
+		err := fmt.Errorf("no 'host' field found in ScaledObject metadata")
+		lggr.Error(err, "ScaledObjectRef", scaledObject)
+		return nil, err
+	}
+	if host == "interceptor" {
+		return &externalscaler.IsActiveResponse{
+			Result: true,
+		}, nil
+	}
+	allCounts := e.pinger.counts()
+	hostCount, ok := allCounts[host]
+	if !ok {
+		err := fmt.Errorf("host '%s' not found in counts", host)
+		lggr.Error(err, "allCounts", allCounts)
+		return nil, err
```

Comment:
If this happens I think the scaler will fail and crash. Can we just set the `hostCount` to 0?

Reply:
The scaler wouldn't crash here, it would just return an error to KEDA. It would be better to keep this logic because if we return an error, KEDA won't take a scaling action. If we return 0, we'll end up telling KEDA to scale the app to zero, which may not be what we want in a failure domain where the scaler doesn't have information about the given host.

Reply:
Agree with that